modules/codegen/src/smithy4s/codegen/v1compat/V1EnumTransformation.scala

```diff
@@ -0,0 +1,75 @@
+package smithy4s.codegen.v1compat
+
+import software.amazon.smithy.build.ProjectionTransformer
+import software.amazon.smithy.build.TransformContext
+import software.amazon.smithy.model.traits.EnumTrait
+import software.amazon.smithy.model.shapes.EnumShape
+import java.util.Optional
+import scala.jdk.CollectionConverters._
+import software.amazon.smithy.model.Model
+
+/**
+  * This is here since the default behavior of smithy v2 with regard to enums is insufficient for our
+  * use case. Specifically, smithy v2 will only transform an EnumTrait to an EnumShape IF it has a `name`
+  * field defined already OR the value matches a certain regex pattern. We want to convert all EnumTraits
+  * to EnumShapes, regardless of what the values are. We are able to do this since we already had logic
+  * that used values as names in the generated code using a toCamelCase function. This transformation
+  * preserves the same logic that existed before, thus causing no changes to the generated code.
+  */
+final class V1EnumTransformation extends ProjectionTransformer {
+  def getName(): String = "V1EnumTransformation"
+
+  @annotation.nowarn("msg=class EnumTrait in package traits is deprecated")
+  def transform(x: TransformContext): Model = {
```

Comment:
One potential con about this transformation is that it drops any traits that existed on the stringshape containing the enumtrait. This makes sense in some cases, such as with a trait that is only allowed on StringShapes and not EnumShapes.. but I imagine it could be problematic too. I'll have to look and see what the smithy core does to address this